browser/ui/views/brave_actions/brave_rewards_action_view.cc

```diff
@@ -84,6 +84,62 @@ const ui::ColorProvider* GetColorProviderForWebContents(
       ui::NativeTheme::GetInstanceForNativeUi()->GetColorProviderKey(nullptr));
 }
 
+// Draws a custom badge for the "verified" checkmark.
+class RewardsBadgeImageSource : public brave::BraveIconWithBadgeImageSource {
+ public:
+  RewardsBadgeImageSource(const gfx::Size& size,
+                          GetColorProviderCallback get_color_provider_callback)
+      : BraveIconWithBadgeImageSource(size,
+                                      std::move(get_color_provider_callback),
+                                      kBraveActionGraphicSize,
+                                      kBraveActionLeftMarginExtra) {}
+
+  void SetBadgeTextAndBackground(const std::string& text, SkColor background) {
+    SetBadge(std::make_unique<Badge>(text, brave::kBadgeTextColor, background));
+    if (background == kBadgeVerifiedBG) {
+      verified_icon_ = true;
+      SetAllowEmptyText(true);
```

Comment:
Nit: I feel like this should be handled in its own function like `SetVerified`, as this function name doesn't imply anything about verification.

Reply:
Good idea - done.